extension/src/popup/helpers/soroban.ts

```diff
@@ -111,15 +111,18 @@ export const getContractDecimals = async (
   contractId: string,
 ) => {
   const contract = new SorobanClient.Contract(contractId);
-  const server = sorobanClient.server;
+
+  if (!sorobanClient.server || !sorobanClient.newTxBuilder) {
+    throw new Error("soroban rpc not supported")
+  }
```

Comment:
well we can't throw it there because there is no way to instantiate a SorobanClient on pubnet/testnet yet so we had to move to an optional server/tx builder in order to allow custom ones as well as support some default networks but not all of them.

Reply:
this was sort of overlooked before because we incorrectly passed in a horizon instance when we were on pubnet/testnet.